tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -191,6 +196,33 @@
    */
   long numBytes();
 
+  /**
+   * Returns the String representation of elements stored in the tensor.
+   *
+   * @param options overrides the default configuration
+   * @return the String representation of the tensor elements
+   * @throws IllegalStateException if this is an operand of a graph
+   */
+  default String dataToString(ToStringOptions... options) {
+    Integer maxWidth = null;
+    if (options != null) {
+      for (ToStringOptions opts : options) {
+        if (opts.maxWidth != null) {
+          maxWidth = opts.maxWidth;
+        }
+      }
+    }
+    return Tensors.toString(this, maxWidth);
+  }
+
+  /**
+   * @param maxWidth the maximum width of the output in characters ({@code null} if unlimited). This
+   *                 limit may surpassed if the first or last element are too long.
+   */
+  static ToStringOptions maxWidth(Integer maxWidth) {
```

Comment:
I'm not sure this method is in the right place. Maybe move it in `ToStringOptions`? Also you need to describe the returned value in the javadoc or the checks might complain.

Reply:
I do it similar to this:
`Layers.Options.create().inputShape(Shape.of(2,2))`
without the ellipsis in the CTORs.